src/utils/eis_utils/src/eis_utils.c

```diff
@@ -418,36 +425,109 @@ EISErr BuildSasCertConnectionString(
     return result;
 }
 
-//
-// External Functions
-//
-
 /**
- * @brief Creates a connection string using the provisioned data within EIS
- * @details Calls into the EIS Identity and Keyservice to create a SharedAccessSignature which is then used
- * to create the connection string, Caller is required to call free() to deallocate the connection string
- * @param[in] expirySecsSinceEpoch the expiration time in seconds since the epoch for the token in the connection string
+ * @brief Writing EIS Identities information to a EIS data file
+ * @details Calls into the EIS Identity and get the identities response, saves to a file
  * @param[in] timeoutMS the timeoutMS in milliseconds for each call to EIS
- * @param[out] provisioningInfo the pointer to the struct which will be initialized with the information for creating a connection to IotHub using the EIS supported provisioning information
- * @returns on success a null terminated connection string, otherwise NULL
+ * returns EISErr value
  */
-EISUtilityResult RequestConnectionStringFromEISWithExpiry(
-    const time_t expirySecsSinceEpoch, uint32_t timeoutMS, ADUC_ConnectionInfo* provisioningInfo)
+EISErr EISIdentitiesPipeWriter(uint32_t timeoutMS, const char* pipePath)
 {
-    EISUtilityResult result = { EISErr_Failed, EISService_Utils };
+    char* identityResponseStr;
 
-    if (provisioningInfo == NULL)
+    EISErr retCode = EISErr_Ok;
+
+    EISErr identityResult = RequestIdentitiesFromEIS(timeoutMS, &identityResponseStr);
+
+    if (identityResponseStr != NULL)
     {
-        result.err = EISErr_InvalidArg;
-        return result;
+        int pipeFd = open(pipePath, O_WRONLY);
+        if (pipeFd < 0)
+        {
+            retCode = EISErr_NamedPipeFailure;
+            goto done;
+        }
+        // dprintf(pipeFd, "%s\n", identityResponseStr);
+        // dprintf(pipeFd, "%d\n", identityResult);
+        write(pipeFd, identityResponseStr, strlen(identityResponseStr));
```

Comment:
why not use the higher-level fprintf followed by a fflush so you can use the string format?